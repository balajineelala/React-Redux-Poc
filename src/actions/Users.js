import fetchUsers from "../services/Users.js";

export function getUsers() {
  return async function getUsersThunk(dispatch) {
    try {
      const response = await fetchUsers();
      dispatch(getUsersSuccess(response.data));
    } catch (error) {
      dispatch(getUsersFailure(error));
    }
  };
}

export const GET_USERS_FAILURE = "GET_USERS_FAILURE";

export function getUsersFailure(error) {
  return {
    type: GET_USERS_FAILURE,
    error,
  };
}

export const GET_USERS_SUCCESS = "GET_USERS_SUCCESS";

export function getUsersSuccess(usersList) {
  return {
    type: GET_USERS_SUCCESS,
    usersList,
  };
}
