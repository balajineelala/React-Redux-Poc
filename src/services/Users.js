import axios from 'axios'

export default function fetchUsers() {
    return axios.get('https://jsonplaceholder.typicode.com/users')
  }