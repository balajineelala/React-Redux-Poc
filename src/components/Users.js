import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { getUsers } from "../actions/Users";
import UsersList from "./UsersList";

class Users extends Component {
  componentDidMount() {
    this.props.getUsers();
  }
  render() {
    const { users } = this.props.users;
    return (
      <div>
        <h1>Users List</h1>
        <UsersList users={users} />
      </div>
    );
  }
}

Users.propTypes = {
  userTitle: PropTypes.string
};

const mapStateToProps = (state) => ({
  users: state.users,
  
});

const mapDispatchToProps = (dispatch) => ({
  getUsers: () => dispatch(getUsers())
});

export default connect(mapStateToProps, mapDispatchToProps)(Users);
