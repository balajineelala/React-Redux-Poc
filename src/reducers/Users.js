import { GET_USERS_SUCCESS, GET_USERS_FAILURE } from "../actions/Users.js";

const initialState = {
  users: []
};

function Users(state = initialState, action) {
  switch (action.type) {
    case GET_USERS_SUCCESS:
      return {
        ...state,
        users: action.usersList
      };
    case GET_USERS_FAILURE:
      return {
        error: action.error
      };
    default:
      return state;
  }
}

export default Users;
