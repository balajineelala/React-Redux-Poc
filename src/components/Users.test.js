import React from 'react';
import Users from './Users';
import promise from "redux-promise-middleware";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { mount } from 'enzyme'
import { Provider } from "react-redux";

const mockStore = configureMockStore([thunk, promise]);

jest.mock("axios");

const initialState = {users : {users: []}}
let store,wrapper 

describe('Users container unit testing',()=>{
   beforeEach(()=>{
    store = mockStore(initialState)
    wrapper = mount((<Provider store={store}><Users /></Provider>))
   })

   //Snap Shot Testing for Connected Components
  it('Users container snapshot testing',()=>{
      expect(wrapper).toMatchSnapshot();
  });

  //checking header h1
  it('contains header - h1', () => {
    expect(wrapper.contains(<h1>User List</h1>)).not.toBe(null)
});

});