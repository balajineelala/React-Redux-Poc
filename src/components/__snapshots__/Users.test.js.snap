// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users container unit testing Users container snapshot testing 1`] = `ReactWrapper {}`;
